api/v1beta1/azuresqlfailovergroup_types.go

```diff
@@ -24,6 +33,7 @@ type AzureSqlFailoverGroupSpec struct {
 	// +kubebuilder:validation:Required
 	Server                       string                          `json:"server"`
 	FailoverPolicy               ReadWriteEndpointFailoverPolicy `json:"failoverPolicy"`
+	// TODO: This field should be a ptr as it must not be specified in the failover policy is Manual
```

Comment:
No because in the case that the failover policy is automatic this field is required, and I believe 0 is a valid value. So you'd be unable to represent the 0 + automatic combination.

I just ignored it when the mode is manual (which is a smell IMO).

I've updated the TODO a bit more to clarify this though.